tests/unit/models/post-test.js

```diff
@@ -153,3 +153,57 @@ test('promiseTypes', function(assert) {
   assert.equal(promiseArray.constructor.superclass, ArrayProxy);
   assert.equal(promiseObject.constructor.superclass, ObjectProxy);
 });
+
+test('model action handles InvalidError', function(assert) {
+  assert.expect(6);
+
+  this.server.put('/posts/:id/publish', (request) => {
+    let data = JSON.parse(request.requestBody);
+    assert.deepEqual(data, { myParam: 'My first param' });
+    assert.equal(request.url, '/posts/1/publish');
+
+    return [422, {}, '{"errors": [{"detail": "First error", "source": {"pointer": "data"}}]}'];
+  });
+
+  let done = assert.async();
+  let payload = { myParam: 'My first param' };
+  let store = this.store();
+  let model = this.subject();
+
+  model.set('id', 1);
+  assert.equal(store.peekAll('post').get('length'), 1);
+
+  model.publish(payload).then((response) => {
```

Comment:
Forget abut this idea :/
> We should refer to them through model.get('errors')... It won't be so easy to do that, but I think this will be a correct way, a compatible solution with the current ember-data flow.

This can't fit into custom actions concept :/ Custom actions should be stateless so we shouldn't kept their errors in the model object :/